File: python/162.find_peak_element.py
from typing import List


class Solution:
    def findPeakElement(self, nums: List[int]) -> int:

        l = 0
        r = len(nums) - 1

        while l < r:
            m = (l + r) // 2

            if m > 0 and nums[m - 1] > nums[m]:
                r = m - 1
            elif m < len(nums) - 1 and nums[m + 1] > nums[m]:
                l = m + 1
            else:
                return m

File: cpp/162.find_peak_element.cpp
#include <vector>

class Solution {
 public:
    int findPeakElement(std::vector<int>& nums) {
        int left = 0;
        int right = nums.size() - 1;

        if (right == 0) {
            return 0;
        }

        while (left <= right) {
            int mid = left + ((right - left) / 2);

            if (mid > 0 && nums[mid - 1] > nums[mid]) {
                right = mid - 1;
            } else if (mid < nums.size() - 1 && nums[mid + 1] > nums[mid]) {
                left = mid + 1;
            } else {
                return mid;
            }
        }
        return 0;
    }
};
